### PR TITLE
docs: remove doc debts and make it up to date per current state of the sdk

### PR DESCRIPTION
**Pull request by denniswon**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR updates migration links, refines schema comments, adds `ClientMiddlewareConfig`, improves documentation clarity, and enhances parameter descriptions.

### Detailed summary
- Updated migration links and sidebar structure
- Refined schema comment precision
- Added `ClientMiddlewareConfig`
- Improved documentation clarity for parameters
- Enhanced parameter descriptions for various actions

> The following files were skipped due to too many changes: `site/packages/aa-core/smart-account-client/actions/checkGasSponsorshipEligibility.md`, `site/packages/aa-core/bundler-client/actions/estimateUserOperationGas.md`, `packages/alchemy/README.md`, `site/snippets/aa-core/send-uo-param.md`, `packages/core/src/types.ts`, `site/resources/terms.md`, `site/packages/aa-core/smart-account-client/actions/buildUserOperation.md`, `site/packages/aa-core/smart-account-client/actions/buildUserOperationFromTx.md`, `site/packages/aa-core/smart-account-client/actions/sendUserOperation.md`, `site/packages/aa-core/smart-account-client/middleware/index.md`, `site/migration-guide.md`, `site/resources/types.md`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: site/packages/aa-core/smart-account-client/actions/upgradeAccount.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ The [`SmartContractClient`](/packages/aa-core/smart-account-client/) instance
 
 - `overrides?:` [`UserOperationOverrides`](/resources/types#useroperationoverrides)
 
-Optional parameter where you can specify override values for `maxFeePerGas`, `maxPriorityFeePerGas`, `callGasLimit`, `preVerificationGas`, `verificationGasLimit`, `paymasterAndData`, or `nonceKey` for the user operation request
+Optional parameter where you can specify override values for `maxFeePerGas`, `maxPriorityFeePerGas`, `callGasLimit`, `preVerificationGas`, `verificationGasLimit`, `paymasterAndData`, or `nonceKey` for the user operation request. You can also specify a `stateOverride` to be passed into `eth_estimateUserOperationGas` during fee estimation.
```
Comment (dphilipson) [inline site/packages/aa-core/smart-account-client/actions/upgradeAccount.md:71]: `eth_estimateUserOperation` doesn't do **fee** estimation, so recommend "fee estimation" -> "gas estimation".

File: site/packages/aa-core/smart-account-client/middleware/index.md [hunk 2]
```diff
@@ -57,9 +59,19 @@ This is a no-op middleware, but you can include any custom step during the user
 
 If you are simulating a user operation using [`simulateUserOperation`](/packages/aa-alchemy/smart-account-client/actions/simulateUserOperation.md), `SmartAccountClient` will include `userOperationSimulator` middleware during the pipeline run to simulate the user operation instead of sending it to the network to be mined.
 
-## Use custom middleware using `ClientMiddlewareConfig` in `SmartAccountClientConfig`
+During the `SmartAccountClient` middleware pipeline run, each middleware is applied to compute certain fields of the [`UserOperationStruct`](/resources/types#useroperationstruct) in the following sequential order:
+
+1. `dummyPaymasterDataMiddleware` -- populates a dummy paymaster data to use in estimation (default: `0x` for no paymater)
```
Comment (dphilipson) [inline site/packages/aa-core/smart-account-client/middleware/index.md:64]: Typo: "paymater" -> "paymaster".

Also I might change "a dummy paymaster data" -> "dummy paymaster data"

File: site/packages/aa-core/smart-account-client/middleware/index.md [hunk 3]
```diff
@@ -57,9 +59,19 @@ This is a no-op middleware, but you can include any custom step during the user
...
+2. `feeEstimator` -- sets maxfeePerGas and maxPriorityFeePerGas
+3. `gasEstimator` -- calls eth_estimateUserOperationGas
```
Comment (dphilipson) [inline site/packages/aa-core/smart-account-client/middleware/index.md:66]: I would say the gas estimator "sets verificationGasLimit, callGasLimit, and preVerificationGas". This middleware doesn't _necessarily_ call eth_estimateUserOperationGas, although if you wanted to mention it you could maybe add "likely by calling eth_estimateUserOperationGas" to the above suggestion.

File: site/packages/aa-core/smart-account-client/middleware/index.md [hunk 4]
```diff
@@ -57,9 +59,19 @@ This is a no-op middleware, but you can include any custom step during the user
...
+4. `customMiddleware` -- default no-op middleware, available for any custom operation to be done in addition and override any of the results returned by previous middlewares
+5. `paymasterMiddleware` -- used to set paymaster fields of a user operation. (default: `0x` for no paymater)
```
Comment (dphilipson) [inline site/packages/aa-core/smart-account-client/middleware/index.md:68]: Typo: "paymater" -> "paymaster"

File: site/packages/aa-core/smart-account-client/middleware/index.md [hunk 5]
```diff
@@ -57,9 +59,19 @@ This is a no-op middleware, but you can include any custom step during the user
...
+6. `userOperationSimulator` -- used for the simulation of a user operation instead of sending it to the network to be mined.
```
Comment (dphilipson) [inline site/packages/aa-core/smart-account-client/middleware/index.md:69]: I'd be interested in a little more detail here. Does this only get used in the case where simulation is happening so it's a conditional part of the pipeline, or does it always run? Is it expected to update the user op somehow or is it returning other information?

File: site/resources/types.md [hunk 6]
```diff
@@ -76,9 +84,17 @@ Each middleware is a function that takes in a user operation object, `UserOperat
...
+An object type that defines the interface for `EntryPoint` functions for packing the user operation to the optimized data structure to enhance performance and reduce gas costs of transactions, and generating the hash of the user operation for the format compatible to the specified `Chain` and `EntryPointVersion`.
+
+::: details EntryPointDef
+<<< @/../packages/core/src/entrypoint/types.ts#EntryPointDef
+:::
+
 ## `Multiplier`
 
-An object type with a required `multipler` field, which is a `number` that must be within the range of 1 to 1000.
+An object type with a required `multipler` field, which is a `number` value with max precision of 4 decmial places.
```
Comment (dphilipson) [inline site/resources/types.md:97]: Typo: "decmial" -> "decimal".

**Review by dphilipson — DISMISSED** [review]
Looks good! A few typos noted inline.

File: site/packages/aa-core/smart-account-client/middleware/index.md (hunk 5 above)
Comment (denniswon) [reply site/packages/aa-core/smart-account-client/middleware/index.md:69]: Added links to where they can learn more (`alchemyUserOperationSimulator`)

**Comment by denniswon** [comment]
updated based on all comments made @dphilipson thanks

**Pull request merged**